cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java

```diff
@@ -393,6 +381,31 @@ private String getMasterPath() {
     return master;
   }
 
+  private File getPodTemplateFile(SparkSubmitContext sparkSubmitContext, boolean isDriver) throws Exception {
+    V1Pod v1Pod = new V1Pod();
+    if (isDriver) {
+      v1Pod.setMetadata(new V1ObjectMetaBuilder()
+                          .addToOwnerReferences(new V1OwnerReferenceBuilder()
+                                                  .withBlockOwnerDeletion(true)
+                                                  .withKind("Pod")
+                                                  .withApiVersion("v1")
+                                                  .withUid(podInfo.getUid())
+                                                  .withName(podInfo.getName()).build()).build());
+    }
+    v1Pod.setSpec(createSparkPodSpec(podInfo, sparkSubmitContext.getLocalizeResources()));
+
+    // Create spark template file. This file gets created in driver pod. We do not delete it because pod will get
```

Comment:
nit: this doesn't actually get created in the driver pod, it gets created in the submitter.

Reply:
So it's not a problem right now when we submit from the workflow, but would it be an issue if we are running a Spark program directly since that gets submitter from app-fabric? Or does it not matter because it's using that new pod runner stuff?

Reply:
Yea, thats a good point. Right now, I noticed driver pods being left around without this change. But we may need to think on what should be owner reference in case we directly submit from appfabric.